virttest/qemu_devices/qcontainer.py

```diff
@@ -2110,6 +2110,8 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
             drv_extra_params = (_.split('=', 1) for _ in
                                 drv_extra_params.split(',') if _)
             for key, value in drv_extra_params:
+                if key == 'cache-size' and imgfmt != 'qcow2':
+                    continue
```

Comment:
As for as know, The `cache-size` just works on `qcow2` now, so the user should have such this background about this.
in other words, should correct it in the case level not in the framework.
I think it's the user's job to decide whether to add the `cache-size` parameter according to the user requirement.
Thanks.

Reply:
Yes, you are right. But for windows guest, although we add \"only qcow2\" in the case level, it will add winutils.iso directly. We have to skip cdrom for windows guest.

Reply:
Hi @XueqiangWei 
yes , I saw the related case: `cache_sizes_test`
My suggestion that modify case line 24 to :
```
    for cache_size, tag in zip(cache_sizes, initial_tag):
        params["drv_extra_params_%s" % tag] = "cache-size=%s" % cache_size
```
to specify the device you wanna set the` cache-size`.
Thanks.

Reply:
Agree with you. It's better to update it in the case level. Thank you very much.